[Cheng Yu Feng] iP

YuFeng0930 opened the pull request.

Commit:
Add Gradle support

Commit:
build.gradle: Update version to 8.29

YuFeng0930 committed:
Level 1. Greet, Echo, Exit

YuFeng0930 committed:
Level 1. Comment

YuFeng0930 committed:
Add chat bot class, list things to do

YuFeng0930 committed:
Add Task class, mark as done

YuFeng0930 committed:
ToDos, Events, Deadlines

YuFeng0930 committed:
Configure testing files

YuFeng0930 committed:
Exception handling

YuFeng0930 committed:
Delete tasks, change mark done a bit

YuFeng0930 committed:
Add comments

YuFeng0930 committed:
Add feature "Save data to hard disk, load data from hard disk"

Add a "FileHandler" class with two functions (read and update).

The file will be updated only when the program terminates instead
of appending tasks to the file when user is using. It might
not be that efficient, but the number of tasks from the user
will not be very large, hence the performance will be okay.

However, the file path specified is not an OS-independent way.
Hence, changes need to be made to let Duke be OS-independent.

YuFeng0930 committed:
Change Deadline and Event String time into LocalDate time

Restrict user input of time element (event, deadline) to format (yyyy-mm-dd).
Format the output of time to (mmm dd yyyy). Handle DateTimeParseException
causing by this new feature.

YuFeng0930 committed:
Change variable name "todo" to "taskList"

This changing is made to clearly distinguish TODO event and todo list.

YuFeng0930 committed:
Merge branch 'master' into branch-Level-8

YuFeng0930 committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/Chatbot.java
#	src/main/java/Deadline.java
#	src/main/java/Event.java

YuFeng0930 committed:
Add class Ui, Parser, Command, TaskList

- Ui: deals with interactions with the user
- Parser: deals with making sense of the user command
- TaskList: contains the task list
- Command: use enum deal with different commands

YuFeng0930 committed:
Package all the classes other than Duke in a package

Since the dependencies of the classes are a bit messy,
all classes other than Duke are package in the same
package for now.

Looking forward to making the dependencies clearer,
and thus have nicer packaging.

YuFeng0930 committed:
Change package name, JUnit test, Create JAR file

Packages are mostly named in lower case, so the package
is renamed to lower case. By doing JUnit test, some exception
handlings are added in class TaskList. Create the JAR file,
and published on github as Duke v0.1.

YuFeng0930 committed:
Add JavaDoc for all non-trivial functions/classes

Some trivial function like constructors or getters do nor have JavaDoc.

YuFeng0930 committed:
Change some lines with more than 100 chars into 2 or more lines

This is to increase the readability of the code.

YuFeng0930 committed:
Add feature "find task by given keyword"

Add one more command "FIND" in command enum, and add one
branch in switch statement. After finding the targeting tasks,
print them out in order.

YuFeng0930 committed:
Merge branch 'branch-A-JavaDoc'

YuFeng0930 committed:
Merge branch 'branch-A-CodingStandard'

YuFeng0930 committed:
Add JavaDoc for function findTask

YuFeng0930 committed:
Revise some JavaDoc comments

For bette readability.

YuFeng0930 committed:
Format switch case statement

Remove the indentation for case statement in switch.

YuFeng0930 committed:
Merge commit 'a75fceeb74708da6ac524c08ef526d38e60ee56b'

YuFeng0930 committed:
Configure address book checkstyle

Add the config file in address book with all the .xml file needed.

YuFeng0930 committed:
Correct bad coding styles detect by

To comply with the coding standard.

YuFeng0930 committed:
Merge branch 'branch-A-CodingStandard'

YuFeng0930 committed:
Configure the basic GUI of Duke

Follow 4 steps on CS2103 website to configure the GUI.
Add classes: DialogBox, Launcher, MainWindow, Main.
Modify build.gradle file for JavaFX.

Haven't yet connect the Duke reply with the GUI.

YuFeng0930 committed:
Connect the chatbot to the GUI

Handle the user input, and give the corresponding duke reply.
Create two new packages: duke.gui, duke.task. Because of the
packaging, the constructor of each subclass of tasks should become
public. Move most of the things into ChatBot class, and Duke class
become a class only be used for CLI.

Modify the return values of functions in TaskList class from void to String,
so that it can be used in handleUserInput function in MainWindow class.
However, the JavaDoc remains the same for now. Hence, a lot of
JavaDoc needs to be changed.

YuFeng0930 committed:
Change GUI background color

Change the background color of MainWindow and Dialog box.
Fix import problems and check naming conventions.

The JavaDoc hasn't been changed yet.

YuFeng0930 committed:
Merge branch 'branch-Level-10'

YuFeng0930 committed:
Add Varargs for parseStringToCommand function in Parser class

markuz5116 commented on src/main/java/duke/Main.java:
```diff
@@ -0,0 +1,32 @@
+    @Override
+    public void start(Stage stage) {
+        try {
+            FXMLLoader fxmlLoader = new FXMLLoader(Main.class.getResource("/view/MainWindow.fxml"));
```
Perhaps FXMLLoader should be camel-cased to FxmlLoader

markuz5116 commented on src/main/java/duke/Main.java:
```diff
@@ -0,0 +1,32 @@
+    public void start(Stage stage) {
+        try {
+            FXMLLoader fxmlLoader = new FXMLLoader(Main.class.getResource("/view/MainWindow.fxml"));
+            AnchorPane ap = fxmlLoader.load();
```
Perhaps the variable name can be more descriptive than ap, so that it will be easier to know what it is down the code.

markuz5116 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,112 @@
+     * @return the list of tasks from previous session
+     */
+    public TaskList readFile() {
+        List<Task> taskList = new ArrayList<>();
```
perhaps using plural form will be clearer.

markuz5116 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,112 @@
+    public void updateFile(TaskList taskList) {
+        String updatedString = "";
+        for (Task t : taskList.getTaskList()) {
+            updatedString += t.getType();
```
I like this variable name, it is clear what it is.

markuz5116 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,112 @@
+     */
+    public void updateFile(TaskList taskList) {
+        String updatedString = "";
+        for (Task t : taskList.getTaskList()) {
```
perhaps t can be more descriptive.

markuz5116 commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,145 @@
+     *
+     * @param newTask the new task to be added
+     */
+    public String addTask(Task newTask) {
```
I like the usage of newTask, it makes it clear that this is not in the tasks.

markuz5116 commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,145 @@
+            if (taskList.size() == 0) {
+                throw new DukeException("There's currently no task in the list.");
+            }
+            List<Task> targetTasks = new ArrayList<>();
```
i like targetTasks, easy to understand that its a lot of targetted tasks.

markuz5116 commented on src/main/java/duke/Chatbot.java:
```diff
@@ -0,0 +1,121 @@
+ */
+public class Chatbot {
+    private final Storage storage;
+    private final TaskList taskList;
```
Personally, I would use tasks rather than taskList, as taskList contains multiple tasks.

markuz5116 commented on src/main/java/duke/Chatbot.java:
```diff
@@ -0,0 +1,121 @@
+            case LIST:
+                return taskList.printTaskList();
+            case DONE:
+                if (taskTypeSplit.length <= 1 || taskTypeSplit[1].isBlank()) {
```
Perhaps it would be good to describe what 1 meant by introducing a constant, maybe DONE_MIN_ARGUMENTS.

markuz5116 commented on src/main/java/duke/Chatbot.java:
```diff
@@ -0,0 +1,121 @@
+        String[] taskTimeSplit;
+        Task newTask;
+        try {
+            String[] taskTypeSplit = input.split(" ");
```
Perhaps calling it inputs could be clearer since you are splitting the input.

markuz5116 replied:
Perhaps declaring what each index of the taskTypeSplit is could be clearer for readers.

markuz5116 commented on src/main/java/duke/Chatbot.java:
```diff
@@ -0,0 +1,121 @@
+
+    }
+
+    public String getDukeReply(String input) {
```
I would personally extract the individual cases codes to different methods such that it will be clearer for readers.

markuz5116 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,32 @@
+     * @param taskString user input
+     * @return corresponding command (enum)
+     */
+    public Command parseStringToCommand(String... taskString) {
```
Perhaps declaring what taskString[0] is could make it clearer for readers what it is being compared to.

markuz5116 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,112 @@
+            File file = new File(filePath);
+            if (!file.createNewFile()) {
+                File myObj = new File(filePath);
+                Scanner myReader = new Scanner(myObj);
```
Do you think that extracting the codes for converting the data file to its tasks could be clearer for the readers?

markuz5116 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,112 @@
+        String updatedString = "";
+        for (Task t : taskList.getTaskList()) {
+            updatedString += t.getType();
+            updatedString += " | ";
```
Perhaps extracting the data representation to specific task classes could make your code clearer.

markuz5116 commented on src/main/java/duke/gui/MainWindow.java:
```diff
@@ -0,0 +1,57 @@
+        dialogContainer.getChildren().add(DialogBox.getDukeDialog(Ui.welcome(), dukeImage));
+    }
+
+    public void setDuke(Chatbot d) {
```
Perhaps a more descriptive variable instead of d can tell readers what the input is?

markuz5116 commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,37 @@
+    @Override
+    public String toString() {
+        return super.toString() + " (by: "
+                + time.format(DateTimeFormatter.ofPattern("MMM d yyyy")) + ")";
```
I like this form of splitting the statement!

markuz5116 commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,37 @@
+        this.time = time;
+    }
+
+    public Event(String name, TaskType type, LocalDate time, boolean done) {
```
Perhaps instead of done, a variable name for the boolean could be better as isDone?

markuz5116 commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,145 @@
+     */
+    public String deleteTask(int order) {
+        try {
+            if (order < 0 || order >= taskList.size()) {
```
Perhaps showing what you are checking for in each case could be clearer?

markuz5116 commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,145 @@
+                throw new DukeException("There's currently no task in the list.");
+            }
+            List<Task> targetTasks = new ArrayList<>();
+            for (Task t : taskList) {
```
Maybe rather than t, using task could be a more descriptive name?

markuz5116 reviewed (commented):
Personally, I think your code is very neat! However, I would extract out some code to make it clearer to readers.

YuFeng0930 committed:
Improve code quality

Rename some plural variable names,
and segment some big functions into smaller functions.

YuFeng0930 committed:
Improve code quality regarding abstraction

Add a new class "Printer" to achieve better abstraction.
Fix the bug of storage and reply message.

YuFeng0930 committed:
Add assertions

Add some assertions to make sure some conditions holds.

YuFeng0930 committed:
Add Streams

Change some for loops into Streams.

YuFeng0930 committed:
Add Lambdas

Change some anonymous function into lambda functions.

YuFeng0930 committed:
Merge pull request #1 from YuFeng0930/branch-A-CodeQuality

Branch a code quality

YuFeng0930 committed:
Merge branch 'master' into branch-A-Assertions

YuFeng0930 committed:
Merge pull request #2 from YuFeng0930/branch-A-Assertions

Add assertions

YuFeng0930 committed:
Merge branch 'master' into branch-A-Streams

YuFeng0930 committed:
Merge pull request #3 from YuFeng0930/branch-A-Streams

Add Streams

YuFeng0930 committed:
Merge branch 'master' into branch-A-Lambdas

YuFeng0930 committed:
Merge pull request #4 from YuFeng0930/branch-A-Lambdas

Add Lambdas

YuFeng0930 committed:
Set up CI

Add gradle.yml

YuFeng0930 committed:
Create gradle.yml

YuFeng0930 committed:
Merge branch 'branch-A-CI'

* branch-A-CI:
  Set up CI

YuFeng0930 committed:
Fix bug

Fix the bug caused by bad merging branch.

YuFeng0930 committed:
Add JavaDoc

Add JavaDoc to those newly added functions.

YuFeng0930 committed:
Fix bug

List command shouldn't have parameter.

YuFeng0930 committed:
Add Sort function based on DateTime

Sort based on DateTime. If a task doesn't have DateTime,
it will be put at the front of task list.

YuFeng0930 committed:
Merge branch 'branch-C-Sort'

* branch-C-Sort:
  Add Sort function based on DateTime

YuFeng0930 committed:
Improve code quality

Modify the constructor of tasks.

YuFeng0930 committed:
Fix gradle.yml file

YuFeng0930 committed:
Beautify the GUI and add JavaDocs

Add background colour for dialog box.
Complete JavaDocs.
Build jar file.

YuFeng0930 committed:
Merge branch 'branch-A-BetterGui'

* branch-A-BetterGui:
  Beautify the GUI and add JavaDocs

YuFeng0930 committed:
Add user guide

YuFeng0930 committed:
Merge branch 'branch-A-UserGuide'

* branch-A-UserGuide:
  Add user guide

YuFeng0930 committed:
Update README.md

YuFeng0930 committed:
Update README.md

YuFeng0930 committed:
Update README.md

YuFeng0930 committed:
Update README.md
